### ISSUE TITLE
Skipper should support SSL/TLS

### ISSUE THREAD

**Issue by Raffo** [zalando/skipper#148]
- It should be possible to turn on and off. 
- It should be possible to specify Key, CA, Certificate.

**Comment by lmineiro** [comment]
Sounds good. What do you think of HTTP/2? We could combine this :)

**Comment by Raffo** [comment]
This [should come out of the box with HTTPS ](https://golang.org/pkg/net/http/)

````
The http package has transparent support for the HTTP/2 protocol when using HTTPS.
````

**Comment by lmineiro** [comment]
That was actually what I meant.

It's probably more test code than anything else

**Comment by hmatinho** [comment]
could you support getting certificates from letsencrypt automatically and managing them ?

**Comment by Raffo** [comment]
@hmatinho I believe this is out of scope for this request. You can maybe open another issue and discuss is there what you would love to see implemented.

**Comment by aryszka** [comment]
@hmatinho  what I would prefer, would be that skipper could monitor the provided path to the certificate and key, and automatically fetch the updates if any from there. And then letsencrypt's certbot standalone, or any other update method, could run separately from skipper, and update the certs in the configured directory. This way we wouldn't create a built-in dependency on a particular service (even if letsencrypt bears all our love :)). Would this approach fit your requirements/workflow?

**Comment by Raffo** [comment]
Closed by #150.
